dspace-api/src/main/java/org/dspace/administer/CreateAdministrator.java

```diff
@@ -114,91 +144,124 @@ protected void negotiateAdministratorDetails()
         String email = null;
         String firstName = null;
         String lastName = null;
-        char[] password1 = null;
-        char[] password2 = null;
         String language = I18nUtil.getDefaultLocale().getLanguage();
+        ConfigurationService cfg = DSpaceServicesFactory.getInstance().getConfigurationService();
+        boolean flag = false;
+        char[] password = null;
+
+
+        if (!line.hasOption('p') && line.hasOption("e")  && line.hasOption("f") && line.hasOption("l")
```

Comment:
The logic here is difficult to understand.  I think that `flag` means "we don't have a password yet" and `dataOK` means "we have all required information other than the password."  It might be clearer to test for these separately.

There are four cases here:  (1) user gave all required values except the password; (2) user gave a password and NONE of the other required values; (3) user gave all required values; (4) [drop off the end of the `if`] some other combination.  I think that we need only three cases, and by separating the password question we need only two.

First set `flag = line.hasOption('p');`.  Then set `dataOK` true if all other required properties are set.  Then `!flag` means "we need to prompt for a password" and `!dataOK` means "we need to prompt for the other required properties."  Is that how you meant it to work?

Reply:
Prompt for password when not provided but all the other fields are provided